services/workflows-service/src/alert/alert.service.ts

```diff
@@ -234,24 +241,72 @@ export class AlertService {
         dataTimestamp: new Date(),
         state: AlertState.triggered,
         status: AlertStatus.new,
-        executionDetails: {},
-        ...Object.assign({}, ...(data || [])),
+        executionDetails: {
+          checkpoint: {
+            hash: computeHash(executionRow),
+            transactionCount: findByKeyCaseInsensitive(executionRow, 'transactionCount'),
+          },
+          executionRow,
+        } as TExecutionDetails,
```

Comment:
This would be more accurate, it will check your code instead of [asserting](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) it.
```suggestion
        } satisfies TExecutionDetails,
```